### ISSUE TITLE
Audit manifest.json and manifest-demo.json

### ISSUE THREAD

**Issue by pdehaan** [mozilla-services/hoverpad#32]
Currently the `version` fields are different, and the `browser_action.default_popup` fields are different. Not sure if [/manifest-demo.json](https://github.com/mozilla-services/hoverpad/blob/master/manifest-demo.json) is even used.

```json
{
  "manifest_version": 2,
  "name": "Hoverpad Demo",
  "version": "1.6",
  "browser_action": {
    "default_title": "Hoverpad",
    "default_popup": "popup/hoverpad.html"
  }
}
```
&mdash; [/manifest-demo.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/manifest-demo.json)

### versus:

```json
{
  "manifest_version": 2,
  "name": "Hoverpad",
  "version": "1.7",
  "browser_action": {
    "default_title": "Hoverpad",
    "default_popup": "index.html"
  }
}
```
&mdash; [/manifest.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/manifest.json)

**Comment by pdehaan** [comment]
Also, not sure if we want to automate copying the `version` field from the package.json file since it's currently defined in:

- [package.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/package.json#L3) &mdash; `"version": "1.7.0",`
- [manifest.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/manifest.json) &mdash; `"version": "1.7",`
- [manifest-demo.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/manifest-demo.json) &mdash; `"version": "1.6",`
- [elm-package.json](https://github.com/mozilla-services/hoverpad/blob/5e9f7bad1795eaa8bf6c9506eb7fae722cb8d4c1/elm-package.json#L2) &mdash; `"version": "1.0.0",`

**Comment by magopian** [comment]
I believe it would make sense indeed to automate the copying. Is there any way to do that?

**Comment by pdehaan** [comment]
> Is there any way to do that?

There _may_ be a package that does that on npm, but I'm not aware of one off the top of my head.

We could probably do some "postversion" npm script hook which calls a "bin/update-version.js" script which copies the package.json `version` field value to `manifest.json`, `manifest-demo.json` (if that's even used anywhere), and `elm-package.json` (if that file's version field needs to be kept in sync).

Only "gotcha"  is that you'd always need to bump the package.json `version` field using `$ npm version [major|minor|patch]` instead of manually setting the package.json `version` value.

**Comment by magopian** [comment]
To be honest I don't think `manifest-demo.json` is used anywhere (ping @Natim), and the `elm-package.json` doesn't need to be kept it sync (it wouldn't hurt, but it's really only useful for releasing a lib). So that would leave the `manifest.json` and `package.json`.

**Comment by Natim** [comment]
I usually use the ``manifest-demo.json`` to build the self-signed extension because it can take a long time before being reviewed on AMO. And I am changing the version number by end before doing so.

**Comment by magopian** [comment]
Can you comment (or document somewhere) how the extensions are built, and (self-)signed please?

**Comment by pdehaan** [comment]
Here is my super-lame ./bin/update-version.js file:

```js
#!/usr/bin/env node

const { writeFileSync } = require('fs');
const PACKAGE_JSON = require('../package.json');
const KEYS = ['version'];

updateFile('../manifest.json');
updateFile('../manifest-demo.json');

function updateFile(file, pkg = PACKAGE_JSON, keys = KEYS) {
  const filePath = require.resolve(file);
  const data = require(filePath);

  keys.forEach(key => data[key] = pkg[key]);
  const str = JSON.stringify(data, null, 2) + '\n';
  writeFileSync(filePath, str);
}
```

**Comment by magopian** [comment]
would that be added to a `script` entry in the `package.json` file?

I'm not sure it's useful at all to be honest, if we have a documented checklist of what to do for a release... what do you think @Natim ?

**Comment by Natim** [comment]
I guess it makes sense to have a way to keep all this version in sync together.

**Comment by magopian** [comment]
Well, if it was automatic, sure, but using a script, it seems to me that the "release checklist" will be:
- update the version `package.json`
- run the `update-version.js` script to mirror this new version in the `manifest.json` and `manifest-demo.json` files

Instead of:
- update the version in the `package.json`, `manifest.json` and `manifest-demo.json` files

I guess we could have a script that would bump the version number in all three files automatically:
- use `bump-version.js <version number>` to increase the version number in all the needed files

**Comment by Natim** [comment]
That's an implementation detail.

If `npm run fx-webext` runs update-version.js then you don't have to care about it at all.

**Comment by pdehaan** [comment]
Yeah, initially I was thinking it could be useful for a `postversion` npm script hook. That way, if you ran `npm version minor`, it'd bump the minor version and then call `postversion` which would sync the other manifest files.

I tried it locally, but npm/git was complaining about a dirty working tree since I had a change in package.json (the added `postversion` script hook). Not sure why npm would care if I had staged files just to bump a version, but yeah. I'd have to look into that more.

This is all probably a super unnecessary optimization at this point. But I do like @Natim's idea of calling it right before `npm run fx-web-ext`.
